rotkehlchen/chain/evm/decoding/curve_lend/decoder.py

```diff
@@ -617,14 +632,110 @@ def _decode_vault_controller_events(self, context: DecoderContext) -> DecodingOu
+
+                paired_events_data = ([event], False)
+                from_event_type = HistoryEventType.SPEND
+                from_event_subtype = HistoryEventSubType.NONE
+                to_event_type = HistoryEventType.SPEND
+                to_event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                to_notes = f'Return {amount} {gauge_asset.symbol} after withdrawing from curve lending vault gauge'  # noqa: E501
+                break
+        else:
+            log.error(f'Failed to find deposit/withdraw event for curve lending vault gauge for {context.transaction}')  # noqa: E501
+
+        return DecodingOutput(action_items=[] if paired_events_data is None else [
+            ActionItem(
+                action='transform',
+                from_event_type=from_event_type,
+                from_event_subtype=from_event_subtype,
+                asset=gauge_asset,
+                amount=amount,
+                to_counterparty=CPT_CURVE,
+                to_notes=to_notes,
+                paired_events_data=paired_events_data,
+                to_event_type=to_event_type,
+                to_event_subtype=to_event_subtype,
+            ),
+        ])
+
+    def _maybe_enrich_curve_lending_gauge_rewards(self, context: EnricherContext) -> TransferEnrichmentOutput:  # noqa: E501
+        """ This enriches rewards paid to users from the vault's gauge contract.
+        It is always a transfer of the reward tokens of the gauge without any other log emitted.
+
+        May raise:
+        - UnknownAsset
+        - WrongAssetType
+        """
+        source_address = bytes_to_address(context.tx_log.topics[1])
+        if (
+                context.event.event_type == HistoryEventType.RECEIVE and
+                context.event.event_subtype == HistoryEventSubType.NONE and
+                source_address in self.gauges and
+                context.event.asset.resolve_to_evm_token().evm_address not in self.vaults
+        ):
+            crypto_asset = context.event.asset.resolve_to_crypto_asset()
+            context.event.event_subtype = HistoryEventSubType.REWARD
+            context.event.notes = f'Receive {context.event.amount} {crypto_asset.symbol} rewards from {self.base.get_or_create_evm_asset(source_address).symbol}'  # noqa: E501
+            context.event.counterparty = CPT_CURVE
+            return TransferEnrichmentOutput(matched_counterparty=CPT_CURVE)
+
+        return FAILED_ENRICHMENT_OUTPUT
+
     # -- DecoderInterface methods
 
     def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
         return (
             dict.fromkeys(self.vaults, (self._decode_vault_events,)) |
-            dict.fromkeys(self.controllers, (self._decode_vault_controller_events,))
+            dict.fromkeys(self.controllers, (self._decode_vault_controller_events,)) |
+            dict.fromkeys(self.gauges, (self._decode_staking_events,))
         )
 
+    def enricher_rules(self) -> list[Callable]:
```

Comment:
is there really no other way other than using an enricher here? We have to absolutely avoid those type of decoders. They are really bad.

Your test transaction is not looking like there is.
https://arbiscan.io/tx/0x8ada83b9904451335617d625c9f9ba255af193be786149c2914a1f81468d85fe

The only check is that if there is a token transfer (can it be any token?) from any ofthe gauges then it is a reward?